### ISSUE TITLE
Segmentation fault (core dumped) when city id is not-continues

### ISSUE THREAD

**Issue by Fingalzzz** [larsenj/AntColonySalesman#1]
`test.txt` has content:

    0	280	133
    1	288	149
    2	288	129
    8	270	133
    4	256	141
    5	256	157

```
root@FINOVO:AntColonySalesman-master# ./tsp test.txt
Segmentation fault (core dumped)
```

**Comment by larsenj** [comment]
Good catch! At a quick glance I think the issue might be with the function for choosing the next city. When my classload dies down a bit I'll revisit this - I want to change it so the cities can be a string instead of an int.
